### PR TITLE
[REVIEW] Bump Numba version to 0.49

### PR DESCRIPTION
**Pull request by gmarkall**
This is needed for the External Memory Management Plugin Interface feature that is new in 0.49, for RMM and CUDF PRs: https://github.com/rapidsai/rmm/pull/317 https://github.com/rapidsai/cudf/pull/5042 

I've tested locally with RMM, cuGraph, cuSpatial, cuDF, Dask-cuDF and cuML - it does not appear to introduce any issues in any of these.

### CODE REVIEW

**Comment by gmarkall** [comment]
Although the Numba 0.49.1 release is finished, it looks like the packages are not available in conda repos for all Python versions yet, hence the fails.

**Comment by gmarkall** [comment]
rerun tests

File: .condarc [hunk 1]
```diff
@@ -4,4 +4,5 @@ channels:
   - rapidsai-nightly
   - nvidia
   - conda-forge
+  - numba
```
Comment (raydouglass) [inline .condarc:7]: We can't use the numba channel

**Comment by gmarkall** [comment]
rerun tests

**Comment by gmarkall** [comment]
The last fail seems to have been down to some HTTP connection fails, so I've started the tests running again. Since it might be a few days until the Anaconda release of Numba 0.49, and this is required by https://github.com/rapidsai/rmm/pull/317, I've reverted back to Numba 0.49.

**Pull request merged**